packages/swap/src/quoting/Leg.ts

```diff
@@ -32,6 +32,14 @@ export default class Leg {
     return this.destAsset !== 'Usdc' ? 'BUY' : 'SELL';
   }
 
+  getBaseAsset(): Exclude<InternalAsset, 'Usdc'> {
+    if (this.destAsset !== 'Usdc') return this.destAsset;
+
+    if (this.srcAsset !== 'Usdc') return this.srcAsset;
+
+    throw new Error('one of the assets must be Usdc');
```

Comment:
```suggestion
    throw new Error('one of the assets must not be Usdc');
```

Reply:
One of them must be `Usdc`. We return the one that isn't